src/SystemCommands-RefactoringSupport/RBPushDownPreview.class.st

```diff
@@ -0,0 +1,121 @@
+"
+I am a preview of push down method refactoring, I can select the methods of class to push down
+"
+Class {
+	#name : #RBPushDownPreview,
+	#superclass : #SpPresenter,
+	#instVars : [
+		'table',
+		'label',
+		'class',
+		'selectedMethods',
+		'selectedAbstractMethods',
+		'refactoring'
+	],
+	#category : #'SystemCommands-RefactoringSupport'
+}
+
+{ #category : #specs }
+RBPushDownPreview class >> defaultSpec [
+	^ SpBoxLayout newVertical
+		add: #label withConstraints: [ :c | c height: self buttonHeight ];
+		add:
+			(SpBoxLayout newHorizontal
+				add: #table;
+				yourself);
+		yourself
+]
+
+{ #category : #examples }
+RBPushDownPreview class >> on: aRefactoring [
+	|temp|
+	temp := self new on: aRefactoring .
+	^ temp openModalWithSpec
+]
+
+{ #category : #examples }
+RBPushDownPreview class >> on: aClass withSelectors: selectors [
+	|temp|
+	temp := self new 
+		on: aClass 
+		withSelectors: selectors asOrderedCollection.
+	^ temp openModalWithSpec
+]
+
+{ #category : #specs }
+RBPushDownPreview class >> title [
+
+	^ 'Pull members down'
+]
+
+{ #category : #subclassresponsibility }
+RBPushDownPreview >> accept [
+	refactoring selectors: (selectedMethods collect: #selector)
+]
+
+{ #category : #'instance creation' }
+RBPushDownPreview >> accept: presenter [
+	self accept.
+	presenter
+				beOk;
+				close
+]
+
+{ #category : #initialization }
+RBPushDownPreview >> initializeChangesTree [
+	table addColumn:
+		((CheckBoxTableColumn title: ' ' evaluated: [ :aClass | 
```

Comment:
Spec 2 use SpCheckBoxTableColum and SpStringTableColumn instead of the equivalent without Sp